component/component.go

```diff
@@ -96,6 +96,13 @@ type Destroyer interface {
 	DestroyFunc() interface{}
 }
 
+// Exec is responsible for starting the exec plugin to allow a deployment
+// plugin to provide it's own exec functionality.
+type Execer interface {
+	// DestroyFunc should return the method handle for the destroy operation.
```

Comment:
Comment is outdated. Given the unique nature of this func in particular, we should try to document what args are available for it (i.e. ExecSessionInfo)